devito/data/allocators.py

```diff
@@ -92,8 +92,12 @@ def initialize(cls):
         return
 
     def alloc(self, shape, dtype, padding=0):
-        datasize = int(reduce(mul, shape))
-        ctype = dtype_to_ctype(dtype)
+        # For complex number, allocate double the size of its real/imaginary part
+        alloc_dtype = dtype(0).real.__class__
+        c_scale = 2 if np.issubdtype(dtype, np.complexfloating) else 1
+
+        datasize = int(reduce(mul, shape) * c_scale)
```

Comment:
potentially PITA comment...

An observation: for complex we implement SoA, while for bundles we implement AoS

it's true we don't support it yet, but one day we might want to be able to allocate bundles-like Functions (hence AoS) in user-land

so, long story short, here I'd  have 

```
datasize  = infer_datasize(shape, dtype)
```

where for now you just put in the logic above

as I said, this is probably a nitpick, so feel free to ignore